paynlpaymentmethods/paynlpaymentmethods.php

```diff
@@ -448,6 +449,7 @@ public function getTransaction($transactionId)
      */
     public function startPayment(Cart $cart, $payment_option_id, $extra_data = array())
     {
+        $this->createPaymentFeeProduct();
```

Comment:
Can you move this line to the addPaymentFee function after the check if the payment fee is >0
https://github.com/paynl/prestashop1.7-plugin/blob/master/paynlpaymentmethods/paynlpaymentmethods.php#L522

Now this is called for every transaction, even if the fee is never used